### PR TITLE
Start negotiate immediately if last one is before debouce interval

### PR DESCRIPTION
**Pull request by cnderrauber**

### CODE REVIEW

**Review by boks1971 — COMMENTED** [review]
Wondering if we can switch to https://github.com/frostbyte73/core for this. Throttle could solve this.

**Comment by cnderrauber** [comment]
> Wondering if we can switch to https://github.com/frostbyte73/core for this. Throttle could solve this.

There will be a deadlock issue if use Throttle here, it excutes the throttled func under its internal lock then the deadlock will happen if `pc.Negotiate` called when `Throttle.excute` running since two goroutines try to acquire two locks with different sequence.

**Comment by boks1971** [comment]
> > Wondering if we can switch to https://github.com/frostbyte73/core for this. Throttle could solve this.
> 
> There will be a deadlock issue if use Throttle here, it excutes the throttled func under its internal lock then the deadlock will happen if `pc.Negotiate` called when `Throttle.excute` running since two goroutines try to acquire two locks with different sequence.

Got it. We can switch later if the callback under lock can be prevented.

cc: @frostbyte73 to check if the function needs to be called under lock there.

File: pkg/rtc/transport.go [hunk 1]
```diff
@@ -1000,23 +1002,22 @@ func (t *PCTransport) Negotiate(force bool) {
 		return
 	}
 
-	if force {
-		t.lock.Lock()
+	var postEvent bool
```
[the remark this answers is not in the document]
Comment (boks1971) [reply pkg/rtc/transport.go:1005]: oh got it, you had to move lock/unlock. Sorry, missed that.

**Pull request merged**